src/lib/krb5/os/locate_kdc.c

```diff
@@ -259,43 +259,26 @@ locate_srv_conf_1(krb5_context context, const krb5_data *realm,
 
         parse_uri_if_https(host, &this_transport, &host, &uri_path);
 
-        /* Find port number, and strip off any excess characters. */
-        if (*host == '[' && (cp = strchr(host, ']')))
-            cp = cp + 1;
-        else
-            cp = host + strcspn(host, " \t:");
-        port = (*cp == ':') ? cp + 1 : NULL;
-        *cp = '\0';
-
-        if (port) {
-            unsigned long l;
-            char *endptr;
-            l = strtoul (port, &endptr, 10);
-            if (endptr == NULL || *endptr != 0)
-                return EINVAL;
-            /* L is unsigned, don't need to check <0.  */
-            if (l > 65535)
-                return EINVAL;
-            port_num = htons(l);
-        } else if (this_transport == HTTPS) {
-            port_num = htons(443);
-        } else {
-            port_num = udpport;
-        }
+        default_port = this_transport == HTTPS ? htons(443) : udpport;
 
-        /* If the hostname was in brackets, strip those off now. */
-        if (*host == '[' && (cp = strchr(host, ']'))) {
-            host++;
-            *cp = '\0';
+        code = k5_parse_host_string(hostlist[i], default_port, &host,
+                                    &port_num);
+        if (code)
+            goto cleanup;
+        port_num = htons(port_num);
+        if (host != NULL) {
```

Comment:
If this happens (someone wrote "kdc = 88" or similar), it seems better to throw an error than silently ignore the entry.

(The current behavior for "kdc = 88" is to try to send to 0.0.0.0 on port 88, which is clearly wrong, but is better than silently ignoring the entry.  Also, right now our error reporting is bad for invalid KDC addresses--"Invalid argument while getting initial credentials"--but we can fix that separately.)